src/Proto.Cluster.Consul/ConsulProvider.cs

```diff
@@ -109,12 +109,18 @@ public async Task ShutdownAsync(bool graceful)
             _logger.LogInformation("Shutting down consul provider");
             //flag for shutdown. used in thread loops
             _shutdown = true;
-            if (!graceful) return;
+
+            if (!graceful)
```

Comment:
Perhaps reverse this check?
If graceful, deregister etc.
And a single log statement at the end